Add date based auditlog filter for auditlogflush command

gwax opened the pull request.

hramezani commented on CHANGELOG.md:
```diff
@@ -1,6 +1,7 @@
 # Changes
 
 #### Improvements
+- feat: add --before-date option to auditlogflush to support retention windows
```
```suggestion
- feat: add `--before-date` option to `auditlogflush` to support retention windows
```

gwax replied:
fixed

hramezani commented on manage.py.tmp:
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python
```
Do we really need this file?

gwax replied:
fixed

hramezani commented:
Thanks @gwax for your patch. I left some comments.
Please rebase your patch and add some tests.

hramezani commented on auditlog/management/commands/auditlogflush.py:
```diff
@@ -8,27 +10,45 @@ class Command(BaseCommand):
+            default=None,
+            help="Flush all entries with a timestamp before a given date (ISO 8601).",
+            dest="before_date",
+            type=lambda b: datetime.datetime.strptime(b, "%Y-%m-%d").date(),
```
```suggestion
            type=datetime.date.fromisoformat,
```

gwax replied:
done

Linkid commented on auditlog/management/commands/auditlogflush.py:
```diff
@@ -8,27 +10,45 @@ class Command(BaseCommand):
+            if before is not None:
+                entries = entries.filter(timestamp__lt=before)
+            else:
+                entries = entries.all()
```
I think you can remove the `else` condition and define `entries` as `LogEntry.objects.all()` directly.

gwax replied:
I'm not sure that I follow. Do you mean to suggest:
```python
entries = LogEntry.objects
if before is not None:
    entries = entries.filter(timestamp__lt=before)
count, _ = entries.delete()
```
or are you suggesting something else?

Linkid replied:
You can write this:
```
entries = LogEntry.objects.all()
if before is not None:
    entries = entries.filter(timestamp__lt=before)
count, _ = entries.delete()
```

gwax replied:
TIL, Django query execution is even lazier than I thought.

Done.

gwax commented:
I have added a separate commit that introduces tests for auditlogflush in the form it exists in master and I have updated my changes to the command to introduce tests for my additions.

hramezani commented on CHANGELOG.md:
```diff
@@ -2,6 +2,7 @@
 
 #### Improvements
 
+- feat: add `--before-date` option to `auditlogflush` to support retention windows
```
```suggestion
- feat: add `--before-date` option to `auditlogflush` to support retention windows ([#365](https://github.com/jazzband/django-auditlog/pull/365))
```

gwax replied:
done

hramezani commented on docs/source/usage.rst:
```diff
@@ -279,12 +279,15 @@ Management commands
 
+You may also specify a date using the ``-b`` of ``--before-date`` option in ISO 8601 format (YYYY-mm-dd) to delete all
+log entries prior to a given date. This may be used to implement time based retention windows.
+
```
```suggestion

.. versionadded:: 2.1.0

```

hramezani reviewed (dismissed):
Thanks @gwax :+1:

hramezani commented:
@gwax Please rebase this PR
